libs/agno/agno/workflow/v2/parallel.py

```diff
@@ -30,40 +41,299 @@ class Parallel:
     name: Optional[str] = None
     description: Optional[str] = None
 
+    def __init__(
+        self,
+        *steps: Union[Step, "Steps", "Loop", "Parallel", "Condition"],
+        name: Optional[str] = None,
+        description: Optional[str] = None,
+    ):
+        self.steps = list(steps)
+        self.name = name
+        self.description = description
+
     def execute(
         self,
         step_input: StepInput,
-        workflow_run_response: WorkflowRunResponse,
         session_id: Optional[str] = None,
         user_id: Optional[str] = None,
     ) -> List[StepOutput]:
-        pass
+        """Execute all steps in parallel using ThreadPoolExecutor"""
+        logger.info(f"Executing {len(self.steps)} steps in parallel: {self.name}")
+
+        def execute_step(step: Step) -> StepOutput:
+            """Execute a single step"""
+            try:
+                return step.execute(step_input, session_id=session_id, user_id=user_id)
+            except Exception as e:
+                logger.error(f"Parallel step {step.name} failed: {e}")
+                # Return error StepOutput instead of raising
+                return StepOutput(
+                    step_name=step.name, content=f"Step {step.name} failed: {str(e)}", success=False, error=str(e)
+                )
+
+        results = []
+        with ThreadPoolExecutor(max_workers=len(self.steps)) as executor:
+            # Submit all tasks
+            future_to_step = {
+                executor.submit(execute_step, step): step for step in self.steps if isinstance(step, Step)
+            }
+
+            # Collect results as they complete
+            for future in as_completed(future_to_step):
+                step = future_to_step[future]
+                try:
+                    result = future.result()
+                    results.append(result)
+                    log_debug(f"Parallel step {step.name} completed")
+                except Exception as e:
+                    logger.error(f"Parallel step {step.name} failed: {e}")
+                    # Add error result
+                    results.append(
+                        StepOutput(
+                            step_name=step.name,
+                            content=f"Step {step.name} failed: {str(e)}",
+                            success=False,
+                            error=str(e),
+                        )
+                    )
+
+        # Sort results by original step order
+        step_name_to_index = {step.name: i for i, step in enumerate(self.steps) if isinstance(step, Step)}
```

Comment:
The steps could be other things than Step? Like Loop, etc.

there must be a better way to keep the order.